data/campaigns/tutorial/scenarios/01_Tutorial_part_1.cfg

```diff
@@ -194,6 +210,85 @@
         [/lua]
     [/event]
 
+    # Give a warning to undo last step if target is out of reach
+    # Check whether at least one of our units can reach any of the target_loc locations
+    [event]
+        name=moveto
+        id=check_undo
+        first_time_only=no
+        [filter]
+            side=1
+        [/filter]
+        [allow_undo][/allow_undo]
+
+        # Store all our units
```

Comment:
Which is the situation where not only the students, but also the fighters should be checked?

Reply:
When you recruit the two elves, one must attack the quintain. In this case either unit on your side may attack and that will enable the end of the current turn. (If that was what you asked.)